src/OpenTK.Core/Platform/Interfaces/IClipboardComponent.cs

```diff
@@ -4,13 +4,65 @@
 using System.Text;
 using System.Threading.Tasks;
 
+#nullable enable
+
 namespace OpenTK.Core.Platform
 {
     // GOAL: We want to support unicode text, rich text and images.
     public interface IClipboardComponent : IPalComponent
     {
+        /// <summary>
+        /// A list of formats that this clipboard component supports.
+        /// </summary>
+        IReadOnlyList<ClipboardFormat> SupportedFormats { get; }
+
+        /// <summary>
+        /// Gets the format of the data currently in the clipboard.
+        /// </summary>
+        /// <returns>The format of the data currently in the clipboard.</returns>
+        ClipboardFormat GetClipboardFormat();
+
+        /// <summary>
+        /// Sets the string currently in the clipboard.
+        /// </summary>
+        /// <param name="text">The text to put on the clipboard.</param>
         void SetClipboardText(string text);
 
-        string GetClipboardText();
+        /// <summary>
+        /// Returns the string currently in the clipboard.
+        /// This function returns null if the current clipboard data doesn't have the <see cref="ClipboardFormat.Text"/> format.
+        /// </summary>
+        /// <returns>The string currently in the clipboard, or null.</returns>
+        string? GetClipboardText();
+
+        /// <summary>
+        /// Gets the audio data currently in the clipboard.
+        /// This function returns null if the current clipboard data doesn't have the <see cref="ClipboardFormat.Audio"/> format.
+        /// </summary>
+        /// <returns>The audio data currently in the clipboard.</returns>
+        // FIXME: Figure out the return type!
+        object? GetClipboardAudio();
```

Comment:
For audio I think a raw PCM buffer and sample rate should suffice in the object structure, however gets very complicated when we also have to consider bit depth and format. Perhaps we should pick single precision floats and roll with it?

Reply:
I think we should go with the most compatible format for OpenAL to allow ease of use with our OpenAL bindings.

Reply:
We will have to take a look at the audio interface for the clipboard at a later date, I think it's fine for now but it will need real world testing to know if it's good.